File: Api/DB/IQuotesRepository.cs
using Api.Entities;

namespace Api.DB;

public interface IQuotesRepository
{
    public Task Create(Quote quote);
    public Task CreateMany(IEnumerable<Quote> quotes);
    public Task<IEnumerable<Quote>> GetAll();
    public Task<Quote?> Get(long id);
    public Task Update(Quote quote);
    public Task Delete(long id);
    public Task WarmUp();
}

File: KotlinApi/src/main/kotlin/no/dunder/mifflin/dundermifflinquotes/db/IQuotesRepository.kt
package no.dunder.mifflin.dundermifflinquotes.db

import no.dunder.mifflin.dundermifflinquotes.domain.dto.Quote

interface IQuotesRepository {
    fun create(quote: Quote)
    fun createMany(quotes: List<Quote>)
    fun getAll(): List<Quote>
    fun get(id: Int): Quote?
    fun update(quote: Quote)
    fun delete(id: Int)
    fun loadCsv()
}